azure_sdk/build.sh

```diff
@@ -0,0 +1,103 @@
+fi
+
+# Apply patches
+for patchfile in "${PREFIX_AZURE_PATCHES}"/*.patch; do
+	if [ ! -f "${PREFIX_AZURE_MARKERS}/$(basename "$patchfile").applied" ]; then
+		echo "applying patch: $patchfile"
+		[[ "${LONG_TEST}" != "y" ]] && [[ $(basename "$patchfile") == "05_c-utility_utests.patch" ]] && continue
+		[[ "${TARGET}" != "armv7m7-imxrt106x"* ]] && [[ "$(basename "$patchfile")" == "04_armv7m7-imxrt106x.patch" ]] && continue
+		patch -d "${PREFIX_AZURE_BUILD}" -p0 -i "$patchfile" && touch "${PREFIX_AZURE_MARKERS}/$(basename "$patchfile").applied"
+	fi
+	# There are some cpp sources, which is currently not supported in Phoenix-RTOS
+	[ "${LONG_TEST}" = "y" ] && rm -rf "${PREFIX_AZURE_SRC}/c-utility/testtools/micromock"
+done
+
+# Set the entered connection string in sample's source (if specified)
+if [ "$AZURE_CONNECTION_STRING" ]; then
+	sed -i "s/\"\[device connection string\]\"/\"${AZURE_CONNECTION_STRING}\"/g" \
+	"${PREFIX_AZURE_SRC}/iothub_client/samples/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample.c" \
+	&& echo "Connection string has been set properly."
+fi
+
+# Set phoenix c compiler and system root
+PHOENIX_COMPILER_CMD=${CC}
+PHOENIX_SYSROOT=$(${CC} --print-sysroot)
+export PHOENIX_COMPILER_CMD
+export PHOENIX_SYSROOT
+
+# Convert ldflags to format recognizable by gcc, for example -q -> -Wl,-q
+LDFLAGS=$(echo " ${LDFLAGS}" | sed "s/\s/,/g" | sed "s/,-/ -Wl,-/g")
+
+# Build (# http and amqp protocols are currently not supported yet)
+# Treat Phoenix-RTOS as Linux and providing toolchain file was the most suitable solution
+(cd "${PREFIX_AZURE_SRC}/build_all/linux" && ./build.sh \
+--toolchain-file "${PREFIX_AZURE}/toolchain-phoenix.cmake" -cl --sysroot="${PHOENIX_SYSROOT}" \
+--no-http --no-amqp)
+
+# Create azure root directory with provided libs, includes and binaries
+mkdir -p "${PREFIX_AZURE_ROOT}" "${PREFIX_AZURE_LIB}" "${PREFIX_AZURE_INC}" "${PREFIX_AZURE_BIN}"
+
+# Copy azure sdk libraries
+(cd "${PREFIX_AZURE_SRC}/cmake/" && \
+echo "${PREFIX_BUILD}/lib" "${PREFIX_AZURE_LIB}" | xargs -n 1 cp "iothub_client/"*.a "deps/umock-c/libumock_c.a" \
+"umqtt/libumqtt.a" "c-utility/libaziotsharedutil.a")
+
+# Copy azure sdk headers
+(cd "${PREFIX_AZURE_SRC}/" && cp -r "iothub_client/inc/." "deps/umock-c/inc/." \
+"deps/azure-macro-utils-c/inc/." "c-utility/inc/." "${PREFIX_AZURE_INC}")
+cp -r "${PREFIX_AZURE_INC}"/* "${PREFIX_BUILD}/include"
+
+# Install iothub client sample and c-utility tests (if built) in file system
+cp "${PREFIX_AZURE_SRC}/cmake/iothub_client/samples/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample" "${PREFIX_AZURE_BIN}"
+b_install "${PREFIX_AZURE_BIN}/iothub_ll_telemetry_sample" /bin
+for dir in "${PREFIX_AZURE_SRC}/cmake/c-utility/tests"/*
+do
+	if [[ "$dir" == *_ut* ]]; then
+		if [[ "$dir" == *crtabstractions_ut ]] || [[ $dir == *httpapicompact_ut ]] || [[ $dir == *uniqueid_ut ]]; then
+			[[ "$dir" == *crtabstractions_ut ]] && test_executable="crt_abstractions_ut_exe"
```

Comment:
- seems like 3 separate if's. - associative array might also be used or a switch-case

```bash
declare -A CUSTOM_DIRS=(
    [crtabstractions_ut]="crt_abstractions_ut_exe"
    [httpapicompact_ut]="httpapi_compact_ut_exe"
    [uniqueid_ut]="uniqueid_ut_linux_exe"
)

for dir in "${PREFIX_AZURE_SRC}/cmake/c-utility/tests"/*; do
    DIR_NAME="$(basename "$dir")"

    if [ -n "${CUSTOM_DIRS[$DIR_NAME]}" ]; then
        trest_executable="${CUSTOM_DIRS[$DIR_NAME]}"
    else
        echo test_executable="${DIR_NAME}_exe"
    fi
    # [...]
done

```